browser: add AlwaysEdit option to ui_defaults

andrei-popovici-dev opened the pull request:
Fixes #13396


Change-Id: I4330facac347da2d863f0a872b91305d01357d2c


* Resolves: #13396
* Target version: master 

### Summary


### TODO

- [x] add ui_default option to start in edit mode

### Checklist

- [x] I have run `make prettier-write` and formatted the code.
- [x] All commits have Change-Id
- [x] I have run tests with `make check`
- [x] I have issued `make run` and manually verified that everything looks okay
- [x] Documentation (manuals or wiki) has been updated or is not required

Minion3665 reviewed (changes requested):
heya!

Nice first draft and thank you very much for getting some code! I have a few questions/problems/adaptations that you should make; please don't be discouraged by this. It's common (and even expected) that your code won't be perfect the first time around - this is our way of helping you polish it so it both fits with everything else and works as expected.

If you need me to go into any more detail on anything I've put here (or on the issue itself) please feel free to ping me. I can probably answer any question you have, and even if I can't, I know people who can... (over the next week I'm on holiday, so you may want to also ping @Darshan-upadhyay1110 for a speedier response)

Thanks again and hope to hear from you again soon, all the best!

andrei-popovici-dev commented:
Hi @Minion3665 !
I have pushed the changes addressing your feedback. I switched to using window.prefs as requested and implemented the one-time auto-start logic (_autoStartConsumed) so the user can still manually exit Edit Mode.

I also noticed that some CI checks failed. Could you please take a look?

Thank you!

Darshan-upadhyay1110 commented on browser/src/control/Permission.js:
```diff
@@ -21,11 +21,14 @@
 
 	setPermission: function (perm) {
 		var button = $('#mobile-edit-button');
+		var alwaysEdit = false;
```
@andrei-popovici-dev check this warning

Darshan-upadhyay1110 commented:
@Minion3665 can you review this patch ?

Minion3665 reviewed (changes requested):
thank you, I like this code a lot better!

I'm happy with your `_autoStartConsumed` solution, I think that's a nice way to do it :) - and thank you very much for using `window.prefs` so this can fit with everything else

I've got a few stylistic comments/nits on the code and then we can look to get this in.

For us to use this, you also need to squash your commits together - I would suggest using [an interactive rebase](https://git-scm.com/docs/git-rebase#_interactive_mode) as described in [this StackOverflow answer](https://stackoverflow.com/questions/5189560/how-do-i-squash-my-last-n-commits-together) to do that. I'd also recommend [rebasing your branch so that it's up to date with our master](https://stackoverflow.com/a/46148801/12293760).

<details>
<summary>Why am I asking for this?</summary>
When we accept code in `CollaboraOnline/online`, we look to have a nice commit history where each commit can be understood on its own. You may be used to contributing to projects where a pull request is polished by adding new commits and at the end of a review you have a nice pull request to refer back to. In `CollaboraOnline/online`, we instead polish commits themselves by pushing updated versions to the pull request - overwriting old versions. For rebasing to stay up-to-date, you seem to have been using GitHub to merge some things but we also can't have those in our `master` branch. Rebasing allows you to keep yourself up-to-date without making extra commits in the history.
</details>

---

Regarding CI - I think it's not your fault. I believe that squashing everything together and rebasing your code will cause CI to rerun (even if there are no actual code changes) so we can see what the status is when you've done that. If it's still failing with the same cause, give me another ping and I'll rerun it and look into it for you...

Thanks again very much, I look forward to hearing back from you!

andrei-popovici-dev committed:
Created startInEditMode ui_defaults option

Signed-off-by: Andrei-Razvan Popovici <popovici.andrei2002@yahoo.com>
Change-Id: I14b1e35155a61cf6063ed7ea16d760afa137cbb5

Darshan-upadhyay1110 reviewed (commented):
hi @andrei-popovici-dev can you please rebase and solve the conflicts on this PR ?

andrei-popovici-dev committed:
Merge conflict resolved

Darshan-upadhyay1110 commented:
Hi, could you please move this PR to Gerrit? [We have now transitioned to Gerrit for code reviews](https://forum.collaboraonline.com/t/your-first-pull-request/41), and I'm currently cleaning up the remaining GitHub PRs.

I'll go ahead and close this PR for now, but we'd be happy to continue the review once it's submitted through Gerrit. Thanks!
